src/ios/CDVSound.m

```diff
@@ -216,14 +216,14 @@ - (void)create:(CDVInvokedUrlCommand*)command
     NSString* mediaId = [command argumentAtIndex:0];
     NSString* resourcePath = [command argumentAtIndex:1];
 
-    CDVAudioFile* audioFile = [self audioFileForResource:resourcePath withId:mediaId doValidation:NO forRecording:NO];
+    CDVAudioFile* audioFile = [self audioFileForResource:resourcePath withId:mediaId doValidation:YES forRecording:NO];
 
     if (audioFile == nil) {
         NSString* errorMessage = [NSString stringWithFormat:@"Failed to initialize Media file with path %@", resourcePath];
         NSString* jsString = [NSString stringWithFormat:@"%@(\"%@\",%d,%@);", @"cordova.require('cordova-plugin-media.Media').onStatus", mediaId, MEDIA_ERROR, [self createMediaErrorWithCode:MEDIA_ERR_ABORTED message:errorMessage]];
         [self.commandDelegate evalJs:jsString];
     } else {
-        NSURL* resourceUrl = [[NSURL alloc] initWithString:resourcePath];
+        NSURL* resourceUrl = audioFile.resourceURL;
```

Comment:
Isn't ```audioFile.resourceURL``` going to always be nil ? (since we are skipping validation in ```audioFileForResource(...)```, which is where we set the resourceURL).